File: app/src/main/java/com/gnoemes/shikimoriapp/entity/manga/data/MangaResponse.java
package com.gnoemes.shikimoriapp.entity.manga.data;

import android.support.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

import org.joda.time.DateTime;

public class MangaResponse {

    @SerializedName("id")
    private long id;
    @SerializedName("name")
    private String name;
    @Nullable
    @SerializedName("russian")
    private String russianName;
    @SerializedName("image")
    private MangaImageResponse image;
    @SerializedName("url")
    private String url;
    @SerializedName("kind")
    private String type;
    @SerializedName("status")
    @Nullable
    private String status;
    @SerializedName("volumes")
    private int volume;
    @SerializedName("chapters")
    private int chapters;
    @SerializedName("aired_on")
    private DateTime airedDate;
    @Nullable
    @SerializedName("released_on")
    private DateTime releasedDate;

    public long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    @Nullable
    public String getRussianName() {
        return russianName;
    }

    public MangaImageResponse getImage() {
        return image;
    }

    public String getUrl() {
        return url;
    }

    public String getType() {
        return type;
    }

    @Nullable
    public String getStatus() {
        return status;
    }

    public int getVolume() {
        return volume;
    }

    public int getChapters() {
        return chapters;
    }

    public DateTime getAiredDate() {
        return airedDate;
    }

    @Nullable
    public DateTime getReleasedDate() {
        return releasedDate;
    }
}

File: app/src/main/java/com/gnoemes/shikimoriapp/entity/manga/data/MangaResponse.kt
package com.gnoemes.shikimoriapp.entity.manga.data

import com.gnoemes.shikimoriapp.entity.anime.domain.Status
import com.gnoemes.shikimoriapp.entity.app.data.LinkedContentResponse
import com.gnoemes.shikimoriapp.entity.common.data.ImageResponse
import com.gnoemes.shikimoriapp.entity.manga.domain.MangaType
import com.google.gson.annotations.SerializedName
import org.joda.time.DateTime

data class MangaResponse(
        @field:SerializedName("id") val id: Long,
        @field:SerializedName("name") val name: String,
        @field:SerializedName("russian") val nameRu: String?,
        @field:SerializedName("image") val image: ImageResponse,
        @field:SerializedName("url") val url: String,
        @field:SerializedName("kind") val type: MangaType,
        @field:SerializedName("status") private val _status: Status?,
        @field:SerializedName("volumes") val volumes: Int,
        @field:SerializedName("chapters") val chapters: Int,
        @field:SerializedName("aired_on") val dateAired: DateTime?,
        @field:SerializedName("released_on") val dateReleased: DateTime?
) : LinkedContentResponse() {
    val status: Status
        get() = _status ?: Status.NONE
}